src/main/bash/sdkman-update.sh

```diff
@@ -50,9 +35,15 @@ function __sdk_update() {
 
 		__sdkman_echo_debug "Fresh and cached candidate lengths: ${#fresh_candidates_csv} ${#SDKMAN_CANDIDATES_CSV}"
 
-		local combined_candidates diff_candidates
+		local fresh_candidates combined_candidates diff_candidates
+
+		if [[ "${zsh_shell}" == 'true' ]]; then
+			fresh_candidates=(${(s:,:)fresh_candidates_csv})
+		else
+			IFS=',' read -a fresh_candidates <<< "${fresh_candidates_csv}"
+		fi
 
-		combined_candidates=("${fresh_candidates[@]}" "${cached_candidates[@]}")
+		combined_candidates=("${fresh_candidates[@]}" "${SDKMAN_CANDIDATES[@]}")
```

Comment:
Despite the counter arguments, I insist on using local variables please. If there are other occurances of this, they should be fixed too. Global variable access is never a good thing, especially if such variables are mutable.

Reply:
Fair points, it would be nice to assign it to a local variable early though if we can't pass it through as a param.